server/services/location.js

```diff
@@ -0,0 +1,65 @@
+import axios from 'axios';
+import GOOGLE_API_KEY from '../config/development';
```

Comment:
So you shouldn't directly import from a config file like this.  The config/index.js file does some magic using the `NODE_ENV` value to determine which config file to load, you can just rely on that.  This allows us to override which configuration file we want to load in which environment.  For example, we may need a specific `test` configuration file, with a dummy value for the GOOGLE_API_KEY, because the tests on this PR are broken - probably by the removal of the development configuration file from the git cache.

Reply:
Here's how it's loaded in the main app: 

```
//
// For sturcture, see config/default.js
console.log('Loading config...');
var config = require('./config');
```

Reply:
Quite honestly the best practice isn't really to use config files _at all_. These values should be passed in and consumed directly from `process.env.*`. For local development, modules like [dotenv](https://www.npmjs.com/package/dotenv) can be used by creating a `.env` file in the project root that's ignored by Git, and can be copy/pasted directly from Lastpass.

Reply:
There isn't consensus on that.  There's quite a bit of debate in the Devops world, especially around containerized applications, about the best way to manage secrets.  Environment variables when dumped into the containers environment are essentially exposed to the whole container.  If someone gets into the container - maybe finds a way to do arbitrary code execution, they have the secrets.  They're available in the environment.

You can secure them somewhat by only putting them in the environment specific to the node process, but that's not easy to do.  That's what we do now for the Ceros container using `aws-env`, but `aws-env` has a rather large security hole in it that could allow for arbitrary code execution, so we're working on removing it.  But then they're also available to all subprocesses of the node process as well.  Meaning, if someone can trigger a tread in your application, or a subprocess, they have them.

If they are placed into a configuration file with appropriate permissions, they can actually significantly less vulnerable than when they're just dumped into the container's environment.  You have to be able to read that specific file in order to access them, as opposed to general access to the environment, or that processes environment.  It's more secure.

Also, from a devops perspective, it's a lot easier to manage.  Getting secrets en-mass into the docker container's environment with out leaking them in some way to the logs is a _royal_ pain in the ass.